### PR TITLE
April Release 34

### PR DESCRIPTION
**Pull request by TCP1231**

### CODE REVIEW

**Comment by solankisamir** [comment]
_We fixed an issue, where the new tenants created by ProvisionTenantV14 and ProvisionTenant15 left the AccountOwnerId column was left null. The procedures have been updated to contain an administrator user._

We can write this in the release note. This is internal implementation.

File: Release 34 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+5. We fixed an issue, where an error was received when creating API schemas with the same namespace.
+6. We fixed an issue, where policies contained all named values which caused a problem for secrets. The policy failed to check if given user had access to secret.
+7. We fixed an issue, whereby Azure Active Directory v1 tokens could be validated, but v2 tokens could not be validated.  With this release, both v1 and v2 tokens can be validated.
+8. Previously, validate-azure-ad-token would only validate tokens produced by the public cloud instance.  With this release, tokens produced in every Azure cloud can be validated.
+9. Previously, the gateway would not refresh the GraphQL schema when it was updated from the Azure portal.  This situation is now fixed.
+10. We implemented a GraphQL context variable to allow access to all portions of the GraphQL context within policy expressions. 
+
+
+### Self-hosted gateway container image release
+1. [2.1.7](https://github.com/Azure/api-management-self-hosted-gateway/releases/tag/Container-v2.1.7)
```
Comment (tomkerkhove) [inline Release 34:19]: Hm, did we update this section because 2.1.8 is out along with 2.2.0-preview-1.

Same Q applies for Helm

File: Release 34 [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+### New features, fixes, and improvements
+
+1. We fixed an issue, where the definition names in the API schema would display blank spaces.
+2. We fixed an issue, where the developer portal was hiding default domain, a parameter has been added to include default domain in the hostnames.
+3. We fixed an issue, where the API revisions were not being properly filtered.
+4. We implemented a fix that will allow users to remove built-in groups, previously where able to add built-in groups but could not remove them.
+5. We fixed an issue, where an error was received when creating API schemas with the same namespace.
+6. We fixed an issue, where policies contained all named values which caused a problem for secrets. The policy failed to check if given user had access to secret.
+7. We fixed an issue, whereby Azure Active Directory v1 tokens could be validated, but v2 tokens could not be validated.  With this release, both v1 and v2 tokens can be validated.
```
Comment (tomkerkhove) [inline Release 34:12]:
```suggestion
7. We fixed an issue, whereby Azure Active Directory v1 tokens could be validated with `validate-azure-ad-token` policy, but v2 tokens could not be validated.  With this release, both v1 and v2 tokens are supported.
```

File: Release 34 [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Release -API Management service: April 2023
+A regular Azure API Management service updated was started February 9, 2023. This release concluded roll out March 31, 2023.
+
+### New features, fixes, and improvements
+
+1. We fixed an issue, where the definition names in the API schema would display blank spaces.
```
Comment (tomkerkhove) [inline Release 34:6]: Let's add this new feature as well
```suggestion
1. Introduce new `context.Deployment.Gateway.*` context properties ([docs](https://learn.microsoft.com/en-us/azure/api-management/api-management-policy-expressions#ContextVariables))
2. We fixed an issue, where the definition names in the API schema would display blank spaces.
```

File: Release 34 [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+2. We fixed an issue, where the developer portal was hiding default domain, a parameter has been added to include default domain in the hostnames.
+3. We fixed an issue, where the API revisions were not being properly filtered.
+4. We implemented a fix that will allow users to remove built-in groups, previously where able to add built-in groups but could not remove them.
+5. We fixed an issue, where an error was received when creating API schemas with the same namespace.
+6. We fixed an issue, where policies contained all named values which caused a problem for secrets. The policy failed to check if given user had access to secret.
```
Comment (mikebudzynski) [inline Release 34:11]: Can you explain what the first sentence means? I don't understand the "policies contained all named values" part.

File: Release 34 [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+7. We fixed an issue, whereby Azure Active Directory v1 tokens could be validated, but v2 tokens could not be validated.  With this release, both v1 and v2 tokens can be validated.
+8. Previously, validate-azure-ad-token would only validate tokens produced by the public cloud instance.  With this release, tokens produced in every Azure cloud can be validated.
```
Comment (mikebudzynski) [inline Release 34:13]: Please remove the double space

File: Release 34 [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+9. Previously, the gateway would not refresh the GraphQL schema when it was updated from the Azure portal.  This situation is now fixed.
```
Comment (mikebudzynski) [inline Release 34:14]: Please remove the double space

Comment (mikebudzynski) [inline Release 34:14]: How about rephrasing to something like:
"The gateway now correctly refreshes the GraphQL schema after it's updated in the Azure portal."

**Review by mikebudzynski — CHANGES_REQUESTED** [review]

File: Release 34 [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Release -API Management service: April 2023
+A regular Azure API Management service updated was started February 9, 2023. This release concluded roll out March 31, 2023.
+
+### New features, fixes, and improvements
+
+1. We fixed an issue, where the definition names in the API schema would display blank spaces.
+2. We fixed an issue, where the developer portal was hiding default domain, a parameter has been added to include default domain in the hostnames.
```
[the remark this answers is not in the document]
Comment (TCP1231) [reply Release 34:9]: https://msazure.visualstudio.com/One/_git/141a6ca5-3ad0-4c0f-9992-e52cc9d7a8f9/pullrequest/7385078

File: Release 34 (hunk 4 above)
Comment (TCP1231) [reply Release 34:13]: https://msazure.visualstudio.com/One/_git/141a6ca5-3ad0-4c0f-9992-e52cc9d7a8f9/pullrequest/7319994

File: Release 34 [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# Release -API Management service: April 2023
+A regular Azure API Management service updated was started February 9, 2023. This release concluded roll out March 31, 2023.
+
+### New features, fixes, and improvements
+
+1. Introduce new `context.Deployment.Gateway.*` context properties ([docs](https://learn.microsoft.com/en-us/azure/api-management/api-management-policy-expressions#ContextVariables))
+2. We fixed an issue, where the definition names in the API schema would display blank spaces.
+2. We fixed an issue, where the definition names in the API schema would display blank spaces.
```
Comment (tomkerkhove) [inline Release 34:8]: Duplicate of previous line?

File: Release 34 (hunk 1 above)
Comment (tomkerkhove) [reply Release 34:21]: Reactivating since it is still outdated

File: Release 34 [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+2. We fixed an issue, where the developer portal was hiding default domain, a parameter has been added to include default domain in the hostnames.
+3. We fixed an issue, where the API revisions were not being properly filtered.
+4. We implemented a fix that will allow users to remove built-in groups, previously where able to add built-in groups but could not remove them.
+5. We fixed an issue, where an error was received when creating API schemas with the same namespace.
+6. We fixed an issue, where policies contained all named values which caused a problem for secrets. The policy failed to check if given user had access to secret.
+7. We fixed an issue, whereby Azure Active Directory v1 tokens could be validated with `validate-azure-ad-token` policy, but v2 tokens could not be validated.  With this release, both v1 and v2 tokens are supported.
+8. Previously, validate-azure-ad-token would only validate tokens produced by the public cloud instance.  With this release, tokens produced in every Azure cloud can be validated.
```
Comment (tomkerkhove) [inline Release 34:15]: The validation is based on Azure AD and not specifically how the token was produced. 

Does this help clarify that? 
```suggestion
8. Previously, validate-azure-ad-token would only validate tokens against the public cloud instance of Azure Active Directory (AD).  With this release, token validation is supported in every Azure cloud where Azure Active Directory is available.
```

**Review by TCP1231 — COMMENTED** [review]
updated

File: Release 34 [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+6. We fixed an issue, where adding API schemas with the same namespace resulted in an error.
+7. We fixed an issue, where policies contained all named values which caused a problem for secrets. The policy failed to check if given user had access to secret.
+8. We fixed an issue, whereby Azure Active Directory v1 tokens could be validated with `validate-azure-ad-token` policy, but v2 tokens could not be validated. With this release, both v1 and v2 tokens are supported.
+9. Previously, validate-azure-ad-token would only validate tokens produced by the public cloud instance. With this release, tokens produced in every Azure cloud can be validated.
+10. Previously, the gateway would not refresh the GraphQL schema when it was updated from the Azure portal. The gateway now correctly refreshes the GraphQL schema after it's updated in the Azure portal.
+11. We implemented a GraphQL context variable to allow access to all portions of the GraphQL context within policy expressions. 
+
+
+### Self-hosted gateway container image release
+1. [2.2.0](https://github.com/Azure/api-management-self-hosted-gateway/releases/tag/Container-v2.2.0-preview-1)
```
[the remark this answers is not in the document]
Comment (TCP1231) [reply Release 34:20]: Ok, i originally didnt list it because it was preview.

File: Release 34 [hunk 11]
```diff
@@ -0,0 +1,28 @@
+# Release - API Management service: February & March 2023
+A regular Azure API Management service update was started February 9, 2023. This release concluded rollout on March 31, 2023.
+
+### New features, fixes, and improvements
+
+1. We add the new `context.Deployment.Gateway.*` object with several properties for accessing gateway information in runtime policies. [Learn more](https://learn.microsoft.com/en-us/azure/api-management/api-management-policy-expressions#ContextVariables).
```
Comment (mikebudzynski) [inline Release 34:6]: "added", not "add"

Comment (mikebudzynski) [inline Release 34:6]: Remove the locale segment from the URL ("/en-us")

**Pull request merged**